support GitLab CI - take three

afeld opened the pull request:
Closes #6. Variation of #7 - more DRY, though I'm not sure how well it would work. Alternative to #4.

@iBobik @tmaier Thoughts?

iBobik committed:
Added example about GitLab CI into README.

afeld committed:
add a Dockerfile for GitLab CI

iBobik commented:
Good point - this is just temporary fix until GitLab team will support non-shell entrypoints. But they told it not be soon. So it is good to make our fix easilly removable - custom Dockerfile and tag on Hub.

So this solution seems the best.

Could you configure Docker Hub to build this branch, so we can test it in our GitLab projects?

afeld committed:
move GitLab CI Dockerfile to its own directory

...so that it can be used by Docker Hub.

/cc https://github.com/docker/hub-feedback/issues/292

afeld commented:
Ok, give [`18fgsa/html-proofer:gitlab-ci`](https://hub.docker.com/r/18fgsa/html-proofer/tags/) a shot.

## Post-merge tasks

* [ ] Change the [build setting](https://hub.docker.com/r/18fgsa/html-proofer/~/settings/automated-builds/) to point to `master` rather than this branch

iBobik commented:
Swapped image in .gitlab-ci.yml and it seems working:

```
Running with gitlab-ci-multi-runner 1.11.1 (a67a225)
  on docker-auto-scale (4e4528ca)
Using Docker executor with image 18fgsa/html-proofer:gitlab-ci ...
Pulling docker image 18fgsa/html-proofer:gitlab-ci ...
Running on runner-4e4528ca-project-880930-concurrent-0 via runner-4e4528ca-machine-1490533928-28063864-digital-ocean-2gb...
Cloning repository...
Cloning into '/builds/janpoboril/mattes.cz'...
Checking out f0a258c6 as master...
Skipping Git submodules setup
$ htmlproofer --empty-alt-ignore
Running ["ScriptCheck", "LinkCheck", "ImageCheck"] on ["."] on *.html... 


Checking 4 external links...
Ran on 5 files!


HTML-Proofer finished successfully.
Job succeeded
```
